### ISSUE TITLE
Minifying a string modifies the receiver

### ISSUE THREAD

**Issue by Phrogz** [rgrove/cssmin#1]
```
irb(main):003:0> x = "s { p : v }"
=> "s { p : v }"
irb(main):004:0> y = CSSMin.minify(x)
=> "s{p:v;}"
irb(main):005:0> x
=> "s{p:v;}"
```

This may be non-obvious if you assume that the call to input.to_s at the top of minify would create a duplicate. String#to_s returns self, not self.dup.

**Comment by rgrove** [comment]
Good catch.

**Comment by rgrove** [comment]
Don't modify the input string. Closed by 4ffdd5c66af2f1772e6f456d1a9f8a010d5695f6
